src/components/uikit/ProgressBar.tsx

```diff
@@ -35,11 +34,11 @@ export function ProgressBar({
 
   const progressAnim = useSharedValue(0);
 
-  useDerivedValue(() => {
+  useEffect(() => {
     progressAnim.value = withTiming(progress, {
       duration: animated ? 200 : 0,
     });
-  }, [step]);
+  }, [animated, progress, progressAnim, step]);
```

Comment:
Hmm this seems confusing as it is not clear when the animation should change as now there are irrelevant deps in the array that should not trigger the change 🤔 What was the problem with the original version?